SportsHub/SportsHub.Api/Controllers/CommentController.cs

```diff
@@ -1,33 +1,99 @@
 ﻿using AutoMapper;
+using FluentValidation;
+using Microsoft.AspNetCore.Authorization;
 using Microsoft.AspNetCore.Mvc;
+using SportsHub.Api.Mapping.Models;
+using SportsHub.Api.Validations;
+using SportsHub.AppService.Authentication.Models.DTOs;
 using SportsHub.AppService.Services;
+using SportsHub.Domain.Constants;
+using SportsHub.Domain.Models;
+using System.Xml.Linq;
 
 namespace SportsHub.Api.Controllers
 {
     [Route("api/[controller]")]
     [ApiController]
     public class CommentController : ControllerBase
     {
-        private readonly ICommentService _service;
+        private readonly ICommentService _commentService;
         private readonly IMapper _mapper;
+        private readonly IValidator<InputCommentDTO> _commentValidator;
+        private readonly IGenerateModelStateDictionary _generateModelStateDictionary;
 
-        public CommentController(ICommentService service, IMapper mapper)
+        public CommentController(ICommentService service, IMapper mapper, IValidator<InputCommentDTO> commentValidator, IGenerateModelStateDictionary generateModelStateDictionary)
         {
-            _service = service;
+            _commentService = service;
             _mapper = mapper;
+            _commentValidator = commentValidator;
+            _generateModelStateDictionary = generateModelStateDictionary;
         }
 
         [HttpGet("GetByArticle")]
-        public async Task<IActionResult> GetByArticleAsync(int articleId)
+        public async Task<ActionResult<IEnumerable<CommentResponseDTO>>> GetByArticleAsync(int articleId)
         {
-            var comments = await _service.GetByArticleAsync(articleId);
+            var comments = await _commentService.GetByArticleAsync(articleId);
 
             if (!comments.Any())
             {
-                return BadRequest($"No comments for this article");
+                return Ok(ValidationMessages.NoCommentsForArticle);
             }
 
-            return Ok(comments);
+            var commentsResponse = _mapper.Map<List<CommentResponseDTO>>(comments);
+
+            return Ok(commentsResponse);
+        }
+
+        [Authorize]
+        [HttpPost("PostComment")]
+        public async Task<ActionResult> PostCommentAsync([FromBody] InputCommentDTO commentInput)
```

Comment:
As for me, it is better to rename `InputCommentDTO` to `CreateCommentRequest`.

Reply:
Updated.